core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/MetricUtils.java

```diff
@@ -106,6 +107,8 @@ public static void collectHttpMetrics(MetricCollector metricCollector, SdkHttpFu
         metricCollector.reportMetric(CoreMetric.HTTP_STATUS_CODE, httpResponse.statusCode());
         httpResponse.firstMatchingHeader("x-amz-request-id")
                     .ifPresent(v -> metricCollector.reportMetric(CoreMetric.AWS_REQUEST_ID, v));
+        httpResponse.firstMatchingHeader(X_AMZN_REQUEST_ID_HEADER)
+                    .ifPresent(v -> metricCollector.reportMetric(CoreMetric.AWS_REQUEST_ID, v));
```

Comment:
Is this meant to be request id header, not request id?

Reply:
It's the header name for request Ids with different format.

Currently, we have two formats I think
```
x-amzn-RequestId
x-amz-request-id
```

Reply:
ok!